.github/workflows/dependency-check.yaml

```diff
@@ -0,0 +1,48 @@
+name: Dependency Check
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  dependency-check:
+    runs-on: ubuntu-20.04
+    steps:
+       # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: create checksum file
+        uses: hypertrace/github-actions/checksum@main
```

Comment:
just a thought, but now that GHA supports composite actions we can probably just create a single shared cache action. They also may have better support for some OOTB caching actions, worth revisiting.

Along the same lines, a single shared dependency check action with all of its setup and reporting should be doable so we can drop in dep check as an independent workflow easily.